### ISSUE TITLE
One to Many BUG

### ISSUE THREAD

**Issue by cdwmhcc** [directus/api#948]
# Bug Report
Using a one-to-many relationship,
select more than one when adding content, 
then submit it, it become one.

## Steps to Reproduce

1. Add Collection `blog_list`
2. Add fileds `title`,`content`
3. Add  Collection `blog`
4. Add fileds `title`,`content`
5. Add fileds `list`,Relational `One to Many`,Relation > Related Collection select `blog_list`
6. Add blog posts,  Click Add New under `blog_list`
7. Repeat adding a few items
8. Save blog posts
9. Edit this blog posts, see it

## Technical Details

- Device: [eg: Desktop, iPhone XS Max, etc]
- OS: [eg: MacOS 10.13.6]
- Web Server: [eg: Nginx 1.8]
- PHP Version: [eg: 7.3]
- Database: [eg: MySQL 5.7]
- Install Method: [eg: cloned `master` branch]

**Comment by bjgajjar** [comment]
Hey @cdwmhcc 

Thanks for the reporting. This is a known issue and we are working on this. Will try to resolve this ASAP.

Just for the confirmation, I hope you have added M2O field in `blog_list` collection.

**Comment by SnowCommerceBrand** [comment]
I can confirm this issue here also The many to many relationship can solve in the temporary but I will attach some diagrams about the issue with a few examples showing what happens after save no column for the one to many relationship seems to be holding the data.
![directus_employee table](https://user-images.githubusercontent.com/37154233/57865541-c5ac9880-77cb-11e9-9cb2-baa116fc4c8e.jpg)


Table not showing relation to the One to Many as column.
![directus](https://user-images.githubusercontent.com/37154233/57865509-b4638c00-77cb-11e9-8e21-2bf0d255d1f4.jpg)


Using a Many to Many Junction Table I was able to relate the data at this point
![manytomany-example](https://user-images.githubusercontent.com/37154233/57865600-e4ab2a80-77cb-11e9-84fe-d85d4e910bee.jpg)

Device: Server
OS: Ubuntu Linux 18
Web Server: Apache 2
PHP Version: 7.1
Database: MySQL 5.7
Install Method: master branch of directus api and frontend

**Comment by bjgajjar** [comment]
Hey @SnowCommerceBrand 

It will be very helpful if you provide your dump `sql` file! If it's private then you can provide it in **Slack** too!

**Comment by cdwmhcc** [comment]
> 嘿@cdwmhcc
> 
> 谢谢你的报道。这是一个已知问题，我们正在努力解决这个问题。将尝试尽快解决此问题。
> 
> 只是为了确认，我希望你在`blog_list`收集中添加了M2O字段。

Can this problem be solved?

**Comment by benhaynes** [comment]
@bjgajjar needs a SQL dump (database export) to properly/quickly replicate and debug this issue. If someone experiencing it can provide that we should be able to get this fixed faster.

**Comment by SnowCommerceBrand** [comment]
Hey @bjgajjar @benhaynes Happy to provide this. Can I send this to you via Slack Private Message?

**Comment by bjgajjar** [comment]
Yeah please send it in message!

**Comment by benhaynes** [comment]
Thank you @SnowCommerceBrand!! 😄

**Comment by bjgajjar** [comment]
Hey @cdwmhcc 

Yeah, It will be fixed soon! 

Hey @SnowCommerceBrand 

Just gone through your DB. You have 2 relational fields `store` (M2O) and `stores2` (O2M)

- `store` has M2O with `client` and `id` is the mapping column (Which is correct)
- `store2` have O2M with `client` and `id` is the mapping column (Which is not correct.)
   When you define O2M relation then the respected collection must have M2O field to map with the 
   O2M relation. You are using the `id` of that collection to map the relation that's why this error occurs.

You can get more information regarding relationships [here](https://docs.directus.io/guides/relationships.html#many-to-one). Let me know if you need any help!

**Comment by bjgajjar** [comment]
The problem occurs as there is an option to add `o2m` field recursively.

Let me describe it through an example. Check the **Steps to Reproduce** of issue. In `Blog` create form we'll have one field named `blog_list` (O2M relation). We have 2 options for that; `Add New` and `Select Existing`. `blog_list` will also have `blog` field inside its collection. From the `Add New` functionality of  `blog_list`; When the user selects `Blog` from there; `blog_list` will be map to the selected `Blog` instead of created one. And if they don't select `blog` from there; then it will be map to the created one only. So we need to disable `M2O` field for relational collection.

Like; If we are adding `Blog` named 'Blog2' and in that add new `blog_list` named 'List1' with selecting 'Blog1'. So when the data saved you will get 'List1' in 'Blog1' instead of 'Blog2'. The same problem may happen with @cdwmhcc. 

For that we can apply one solution; if there is `O2M` field and the user tries to use `Add New` functionality then we have to disable `M2O` field from the relational collection. Thus in our scenario when the user tries to add `blog_list` they'll not find `blog`.

Let me know your thoughts @benhaynes

**Comment by rijkvanzanten** [comment]
Disabling the m2o field that links back to the o2m in the modal of the o2m should take care of this right @bjgajjar?

**Comment by bjgajjar** [comment]
Yes! Exactly. So I think this can be considered as an APP issue instead of the API!
